packages/at_secondary_server/lib/src/verb/handler/abstract_verb_handler.dart

```diff
@@ -152,4 +152,27 @@ abstract class AbstractVerbHandler implements VerbHandler {
       return false;
     }
   }
+
+
+  /// This function checks the validity of a provided OTP.
+  /// It returns true if the OTP is valid; otherwise, it returns false.
+  /// If the OTP is not found in the keystore, it also returns false.
+  ///
+  /// Additionally, this function removes the OTP from the keystore to prevent its reuse.
+  Future<bool> isOTPValid(String? otp) async {
+    if (otp == null) {
+      return false;
+    }
+    String otpKey =
+        'private:${otp.toLowerCase()}${AtSecondaryServerImpl.getInstance().currentAtSign}';
+    AtData otpAtData;
+    try {
+      otpAtData = await keyStore.get(otpKey);
+    } on KeyNotFoundException {
+      return false;
+    }
+    // Remove the key from keystore to prevent reuse of OTP.
+    await keyStore.remove(otpKey);
```

Comment:
Is this safe?

This method is called in two places
1. EnrollVerbHandler._handleEnrollmentRequest : If the server is restarted after this method has been called, but before the handling of of the enrollment request has completed, then the OTP will have been deleted and when the client retries the OTP will no longer exist. Feels like the decision to remove the OTP ought to be made by the enrollment verb handler as its final piece of processing ...?
2. OtpVerbHandler for the `validate` operation : What is the purpose of the `validate` operation? Who calls it in what circumstances and why?